### PR TITLE
Switch to current user profile on avatar click

### PR DESCRIPTION
**Pull request by akalin-keybase**
...on sidebar.

### CODE REVIEW

File: shared/app/nav.desktop.js [hunk 1]
```diff
@@ -49,11 +49,18 @@ const mapStateToProps = (state: TypedState, ownProps: OwnProps) => ({
 })
 
 const mapDispatchToProps = (dispatch: Dispatch, ownProps: OwnProps) => ({
-  switchTab: (tab: Tab) => {
+  switchTab: (tab: Tab, username?: string) => {
```
Comment (akalin-keybase) [inline shared/app/nav.desktop.js]: Not sure if there's a better way to get the username...

**Comment by akalin-keybase** [comment]
Addressed comments, PTAL @chrisnojima

**Comment by akalin-keybase** [comment]
oh hmm, I should take a look at the mobile side, too...

**Comment by akalin-keybase** [comment]
Ah, nothing to do since there's no avatar tab!

**Comment by keybase-ci-visdiff** [comment]
The commits 1dbc8d631568...beb9a78e4f1e introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Profile-Your_Profile_Broken** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1502232146141-1dbc8d631568-beb9a78e4f1e-linux/Profile-Your_Profile_Broken-1dbc8d631568.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1502232146141-1dbc8d631568-beb9a78e4f1e-linux/Profile-Your_Profile_Broken-beb9a78e4f1e.png)  
   Profile-Your_Profile_Broken comparison
</summary>

File: shared/app/nav.desktop.js [hunk 2]
```diff
@@ -54,6 +54,17 @@ const mapDispatchToProps = (dispatch: Dispatch, ownProps: OwnProps) => ({
       // clicking the chat tab when already selected should do nothing.
       return
     }
+
+    // If we're going to the profile tab, switch to the current user's
+    // profile first before switching tabs, if necessary.
+    if (tab === profileTab) {
+      dispatch(navigateTo([], [profileTab]))
+      if (ownProps.routeSelected !== profileTab) {
```
Comment (chrisnojima) [inline shared/app/nav.desktop.js]: i think you can just do this and it'll be benign if the route is the same right?

Comment (akalin-keybase) [inline shared/app/nav.desktop.js]: Yeap! Done.

**Pull request merged**